napari/layers/image/image.py

```diff
@@ -163,12 +164,19 @@ def __init__(
         )
 
         self.events.add(
-            interpolation=Event, rendering=Event, iso_threshold=Event
+            interpolation=Event,
+            rendering=Event,
+            iso_threshold=Event,
+            complex_func=Event,
         )
 
         # Set data
         self.is_pyramid = is_pyramid
         self.rgb = rgb
+        self.iscomplex = (
```

Comment:
can this be `is_complex` - I think we use `is_pyramid`, so seems more consistent